src/main/java/jenkins/plugins/http_request/HttpRequestStep.java

```diff
@@ -0,0 +1,292 @@
+    private String validResponseCodes         = DescriptorImpl.validResponseCodes;
+    private String validResponseContent       = DescriptorImpl.validResponseContent;
+    private MimeType acceptType               = DescriptorImpl.acceptType;
+    private MimeType contentType              = DescriptorImpl.contentType;
+    private Integer timeout                   = DescriptorImpl.timeout;
+    private Boolean consoleLogResponseBody    = DescriptorImpl.consoleLogResponseBody;
+    private String authentication             = DescriptorImpl.authentication;
+    private List<NameValuePair> customHeaders = DescriptorImpl.customHeaders;
+
+    @DataBoundConstructor
+    public HttpRequestStep(String url) {
+        this.url = url;
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    @DataBoundSetter
+    public void setHttpMode(HttpMode httpMode) {
+        this.httpMode = httpMode;
+    }
+
+    public HttpMode getHttpMode() {
+        return httpMode;
+    }
+
+    @DataBoundSetter
+    public void setValidResponseCodes(String validResponseCodes) {
+        this.validResponseCodes = validResponseCodes;
+    }
+
+    public String getValidResponseCodes() {
+        return validResponseCodes;
+    }
+
+    @DataBoundSetter
+    public void setValidResponseContent(String validResponseContent) {
+        this.validResponseContent = validResponseContent;
+    }
+
+    public String getValidResponseContent() {
+        return validResponseContent;
+    }
+
+    @DataBoundSetter
+    public void setAcceptType(MimeType acceptType) {
+        this.acceptType = acceptType;
+    }
+
+    public MimeType getAcceptType() {
+        return acceptType;
+    }
+
+    @DataBoundSetter
+    public void setContentType(MimeType contentType) {
+        this.contentType = contentType;
+    }
+
+    public MimeType getContentType() {
+        return contentType;
+    }
+
+    @DataBoundSetter
+    public void setTimeout(Integer timeout) {
+        this.timeout = timeout;
+    }
+
+    public Integer getTimeout() {
+        return timeout;
+    }
+
+    @DataBoundSetter
+    public void setConsoleLogResponseBody(Boolean consoleLogResponseBody) {
+        this.consoleLogResponseBody = consoleLogResponseBody;
+    }
+
+    public Boolean getConsoleLogResponseBody() {
+        return consoleLogResponseBody;
+    }
+
+    @DataBoundSetter
+    public void setAuthentication(String authentication) {
+        this.authentication = authentication;
+    }
+
+    public String getAuthentication() {
+        return authentication;
+    }
+
+    @DataBoundSetter
+    public void setCustomHeaders(List<NameValuePair> customHeaders) {
+        this.customHeaders = customHeaders;
+    }
+
+    public List<NameValuePair> getCustomHeaders() {
+        return customHeaders;
+    }
+
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl) super.getDescriptor();
+    }
```

Comment:
Making it `static` is probably simpler and clearer.

For cases where you actually need an instance method on a descriptor, for example because you are looking up global configuration state, it is best to use

```java
Jenkins.get[Active]Instance().getDescriptorByType(TheDescribable.DescriptorImpl.class).yourMethod();
```